site/compiler/test.js

```diff
@@ -133,14 +149,324 @@ describe('site/compiler', () => {
           'upcoming-event': upcomingEvent,
           'designing-in-cross-functional-teams': designingEvent,
         },
+        badgers: [],
+        categories: [],
       });
 
-      expect(pages.length).to.equal(10);
+      expect(pages.length).to.equal(11);
       expect(pages[7].body).to.match(/No internet connection/);
       expect(pages[8].path).to.equal('about-us/events/2017/01/31/upcoming-event/index.html');
       expect(pages[8].body).to.match(/Upcoming Event/);
       expect(pages[9].path).to.equal('about-us/events/2016/08/03/designing-in-cross-functional-teams/index.html');
       expect(pages[9].body).to.match(/Designing in cross-functional teams/);
     });
+
+    it('renders the dynamic badger pages of the site', () => {
+      const pages = compileSite({
+        jobs: [],
+        job: {},
+        contactUsURL: '',
+        featuredBlogPosts: [],
+        events: [],
+        event: {},
+        badgers: [{
+          firstName: 'Alex',
+          lastName: 'Savin',
+          categories: [
+            'Engineering',
+            'Leadership',
+          ],
+        }],
+        categories: ['Engineering', 'Leadership'],
```

Comment:
We could actually use my library to go from flowTypes -> test data here? https://github.com/JoeStanton/babel-transform-flow-introspection/blob/master/use-cases/generative-testing.js